provider/provider.go

```diff
@@ -0,0 +1,174 @@
+
+func NewAwsProvider(ctx context.Context, configPath, controllerID string) (execution.ExternalProvider, error) {
+	conf, err := config.NewConfig(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("error loading config: %w", err)
+	}
+	awsCli, err := client.NewAwsCli(ctx, conf)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get AWS CLI: %w", err)
+	}
+
+	return &AwsProvider{
+		cfg:          conf,
+		controllerID: controllerID,
+		awsCli:       awsCli,
+	}, nil
+}
+
+type AwsProvider struct {
+	cfg          *config.Config
+	controllerID string
+	awsCli       *client.AwsCli
+}
+
+func (a *AwsProvider) CreateInstance(ctx context.Context, bootstrapParams params.BootstrapInstance) (params.ProviderInstance, error) {
+	if bootstrapParams.OSArch != params.Amd64 {
+		return params.ProviderInstance{}, fmt.Errorf("unsupported architecture: %s", bootstrapParams.OSArch)
+	}
+
+	spec, err := spec.GetRunnerSpecFromBootstrapParams(a.cfg, bootstrapParams, a.controllerID)
+	if err != nil {
+		return params.ProviderInstance{}, fmt.Errorf("failed to get runner spec: %w", err)
+	}
+
+	instanceID, err := a.awsCli.CreateRunningInstance(ctx, spec)
+	if err != nil {
+		return params.ProviderInstance{}, fmt.Errorf("failed to create instance: %w", err)
+	}
+
+	instance := params.ProviderInstance{
+		ProviderID: instanceID,
+		Name:       spec.BootstrapParams.Name,
+		OSType:     spec.BootstrapParams.OSType,
+		OSArch:     spec.BootstrapParams.OSArch,
+		Status:     "running",
+	}
+
+	return instance, nil
+
+}
+
+func (a *AwsProvider) DeleteInstance(ctx context.Context, instance string) error {
+	var inst string
+	if strings.HasPrefix(instance, "i-") {
+		inst = instance
+	} else {
+		tags := map[string]string{
+			"GARM_CONTROLLER_ID": "",
+			"Name":               instance,
+		}
+
+		tmp, err := a.awsCli.FindInstanceByTags(ctx, tags)
+		if err != nil {
+			if errors.Is(err, garmErrors.ErrNotFound) {
+				return nil
+			}
+			return fmt.Errorf("failed to determine instance: %w", err)
+		}
+		inst = *tmp.InstanceId
+	}
+
+	if inst == "" {
+		return nil
+	}
+
+	if err := a.awsCli.TerminateInstance(ctx, inst); err != nil {
+		return fmt.Errorf("failed to terminate instance: %w", err)
+	}
+
+	return nil
+}
+
+func (a *AwsProvider) GetInstance(ctx context.Context, instance string) (params.ProviderInstance, error) {
+	awsInstance, err := a.awsCli.GetInstance(ctx, instance)
+	if err != nil {
+		return params.ProviderInstance{}, fmt.Errorf("failed to get VM details: %w", err)
+	}
+	if awsInstance == nil {
+		return params.ProviderInstance{}, nil
+	}
+
+	providerInstance := params.ProviderInstance{
+		ProviderID: *awsInstance.InstanceId,
+		Name:       *awsInstance.Tags[0].Value,
+		Status:     params.InstanceStatus(awsInstance.State.Name),
+		OSType:     params.OSType(awsInstance.Platform),
+		OSArch:     params.OSArch(awsInstance.Architecture),
+		OSVersion:  *awsInstance.PlatformDetails,
+	}
+	return providerInstance, nil
```

Comment:
I think we need a new function here that translates EC2 instances to GARM instance info. Something like:


```go
func awsInstanceToParamsInstance(ec2Instance types.Instance) (params.ProviderInstance, error) {
	if ec2Instance.InstanceId == nil {
		return params.ProviderInstance{}, fmt.Errorf("instance ID is nil")
	}
	details := params.ProviderInstance{
		ProviderID: *ec2Instance.InstanceId,
	}

	for _, tag := range ec2Instance.Tags {
		if tag.Key == nil || tag.Value == nil {
			continue
		}
		switch *tag.Key {
		case "Name":
			details.Name = *tag.Value
		case "OSType":
			details.OSType = params.OSType(*tag.Value)
		case "OSArch":
			details.OSArch = params.OSArch(*tag.Value)
		}
	}

	switch ec2Instance.State.Name {
	case types.InstanceStateNameRunning,
		types.InstanceStateNameShuttingDown,
		types.InstanceStateNameStopping:

		details.Status = params.InstanceRunning
	case types.InstanceStateNameStopped,
		types.InstanceStateNameTerminated:

		details.Status = params.InstanceStopped
	default:
		details.Status = params.InstanceStatusUnknown
	}
	return details, nil
}
```

This function could be used both here and in `ListInstances()`